addons/pos_loyalty/static/src/overrides/models/pos_order.js

```diff
@@ -1467,7 +1484,7 @@ patch(PosOrder.prototype, {
                 // Allow rejecting a gift card that is not yet paid.
                 const program = this.models["loyalty.program"].get(payload.program_id);
                 if (program && program.program_type === "gift_card" && !payload.has_source_order) {
-                    const confirmed = await ask(this.env.services.dialog, {
+                    const confirmed = await ask(this.pos.env.services.dialog, {
```

Comment:
You should never use services from templates. You can return the error and open the service at the point where the method was called.